src/lib/mappers/company-address.js

```diff
@@ -0,0 +1,22 @@
+const CompanyAddress = require('../models/company-address');
+const DateRange = require('../models/date-range');
+const Address = require('../models/address');
+
+const addressMapper = require('./address');
+
+const crmToModel = row => {
+  const companyAddress = new CompanyAddress(row.companyAddressId);
+  companyAddress.fromHash({
+    dateRange: new DateRange(row.startDate, row.endDate),
+    companyId: row.companyId,
+    isDefault: row.isDefault
+  });
+
+  if (row.role) companyAddress.roleName = row.role.name;
+  if (row.address) companyAddress.address = addressMapper.crmToModel(row.address);
+  if (row.addressId) companyAddress.address = new Address(row.addressId);
```

Comment:
It may be possible that on the CRM end, when doing a bookshelf call withRelated models, that it responds with both `address` and `addressId` properties populated.
In this case I think the address from the line above would get overwritten - maybe we need:
```
if (row.address) {
  companyAddress.address = addressMapper.crmToModel(row.address);
}
else if (row.addressId) companyAddress.address = new Address(row.addressId);
```